test/test-lsp-server.cpp

```diff
@@ -286,6 +286,71 @@ TEST_F(test_linting_lsp_server, opening_document_lints) {
   EXPECT_THAT(this->lint_calls, ElementsAre(u8"let x = x;"));
 }
 
+TEST_F(test_linting_lsp_server, opening_document_language_id_js_lints) {
+  this->lint_callback = [&](configuration&, padded_string_view code,
+                            ::simdjson::ondemand::value& uri,
+                            ::simdjson::ondemand::value& version,
+                            byte_buffer& notification_json) {
+    EXPECT_EQ(code, u8"let x = x;");
+    EXPECT_EQ(simdjson_to_jsoncpp(uri), "file:///test.js");
+    EXPECT_EQ(simdjson_to_jsoncpp(version), 10);
```

Comment:
Blocking: `uri` and `version` changed to `string8_view`. This is the cause of your build failures:
```suggestion
                            string8_view uri_json,
                            string8_view version_json,
                            byte_buffer& notification_json) {
    EXPECT_EQ(code, u8"let x = x;");
    EXPECT_EQ(uri_json, u8"\"file:///test.js\"");
    EXPECT_EQ(version_json, u8"10");
```